File: mydir/src/Entity/Evenement.php
<?php

namespace App\Entity;

use App\Repository\EvenementRepository;
use Doctrine\Common\Collections\ArrayCollection;
use Doctrine\Common\Collections\Collection;
use Doctrine\ORM\Mapping as ORM;

/**
 * @ORM\Entity(repositoryClass=EvenementRepository::class)
 */
class Evenement
{
    /**
     * @ORM\Id
     * @ORM\GeneratedValue
     * @ORM\Column(type="integer")
     */
    private $id;

    /**
     * @ORM\Column(type="string", length=255)
     *
     */
    private $libelle;

    /**
     * @ORM\Column(type="string", length=255)
     */
    private $description;

    /**
     * @ORM\Column(type="string", length=255)
     */
    private $DateDebut;

    /**
     * @ORM\Column(type="string", length=255)
     */
    private $DateFin;

    /**
     * @ORM\Column(type="integer")
     */
    private $PrixInscription;

    /**
     * @ORM\ManyToMany(targetEntity=Promotion::class, inversedBy="evenements")
     */
    private $promotions;

    public function __construct()
    {
        $this->promotions = new ArrayCollection();
    }



    public function getId(): ?int
    {
        return $this->id;
    }

    public function getLibelle(): ?string
    {
        return $this->libelle;
    }

    public function setLibelle(string $libelle): self
    {
        $this->libelle = $libelle;

        return $this;
    }

    public function getDescription(): ?string
    {
        return $this->description;
    }

    public function setDescription(string $description): self
    {
        $this->description = $description;

        return $this;
    }

    public function getDateDebut(): ?string
    {
        return $this->DateDebut;
    }

    public function setDateDebut(string $DateDebut): self
    {
        $this->DateDebut = $DateDebut;

        return $this;
    }

    public function getDateFin(): ?string
    {
        return $this->DateFin;
    }

    public function setDateFin(string $DateFin): self
    {
        $this->DateFin = $DateFin;

        return $this;
    }

    public function getPrixInscription(): ?int
    {
        return $this->PrixInscription;
    }

    public function setPrixInscription(int $PrixInscription): self
    {
        $this->PrixInscription = $PrixInscription;

        return $this;
    }

    /**
     * @return Collection|Promotion[]
     */
    public function getPromotions(): Collection
    {
        return $this->promotions;
    }

    public function addPromotion(Promotion $promotion): self
    {
        if (!$this->promotions->contains($promotion)) {
            $this->promotions[] = $promotion;
        }

        return $this;
    }

    public function removePromotion(Promotion $promotion): self
    {
        $this->promotions->removeElement($promotion);

        return $this;
    }
}

File: JavaFX3A22/src/Edu/esprit/entities/Evenement.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package Edu.esprit.entities;

import java.sql.Date;
import java.time.LocalDate;

/**
 *
 * @author 21652
 */
public class Evenement {
       private Integer id ;
    private String libelle ;
    private String description;
    private int PrixInscription;
    private String datefin;
    private String datedebut;

    public Evenement() {
    }

    public Evenement(Integer id, String libelle, String description) {
        this.id = id;
        this.libelle = libelle;
        this.description = description;
    }

    public Evenement(String libelle, String description, int PrixInscription, String datefin, String datedebut) {
        this.libelle = libelle;
        this.description = description;
        this.PrixInscription = PrixInscription;
        this.datefin = datefin;
        this.datedebut = datedebut;
    }
     public Evenement(Integer id,String libelle, String description,String datedebut, String datefin, int PrixInscription) {
        this.id = id;
         this.libelle = libelle;
        this.description = description;
        this.PrixInscription = PrixInscription;
        this.datefin = datefin;
        this.datedebut = datedebut;
    }

    public Evenement(Integer id, String libelle, String description, int PrixInscription, String datefin, String datedebut) {
        this.id = id;
        this.libelle = libelle;
        this.description = description;
        this.PrixInscription = PrixInscription;
        this.datefin = datefin;
        this.datedebut = datedebut;
    }

  

    public Integer getId() {
        return id;
    }

    public String getLibelle() {
        return libelle;
    }

    public String getDescription() {
        return description;
    }

    public int getPrixInscription() {
        return PrixInscription;
    }

    public String getDatefin() {
        return datefin;
    }

    public String getDatedebut() {
        return datedebut;
    }

    public void setId(Integer id) {
        this.id = id;
    }

    public void setLibelle(String libelle) {
        this.libelle = libelle;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public void setPrixInscription(int PrixInscription) {
        this.PrixInscription = PrixInscription;
    }

    public void setDatefin(String datefin) {
        this.datefin = datefin;
    }

    public void setDatedebut(String datedebut) {
        this.datedebut = datedebut;
    }

    @Override
    public String toString() {
        return "Evenement{" + "id=" + id + ", libelle=" + libelle + ", description=" + description + ", PrixInscription=" + PrixInscription + ", datefin=" + datefin + ", datedebut=" + datedebut + '}';
    }
    
}
    




